Improvements

adalon opened the pull request.

adalon committed:
Update Microsoft.Build packages

To fix MsBuild::IsRunningFromVisualStudio usage by the core msbuild targets

adalon committed:
Added build targets to be included in the nuget package

Which provide the FormatCode/FormatXml targets. This allow the clients
to specify just the following:

  <ItemGroup>
    <FormatSolution Include="$(SourceDir)**\*.sln" />

    <FormatXml Include="$(SourceDir)**\*.xml" />
    <FormatXml Include="$(SourceDir)**\*.xaml" />
  </ItemGroup>

kzu commented on src/build/CodeFormatter.targets:
```diff
@@ -0,0 +1,45 @@
+        if(UseTabs)
+        {
+        settings.Indent = true;
+        settings.IndentChars = "\t";
```
This should use the editorconfig setting for the given file file extension, rather than a hardcoded tab...

adalon committed:
Added support for .editorConfig

Added support to use by default the .editorConfig file to configure the roslyn
workspace formatting options.

Added /noEditorConfig argument if the user does not want to use it.

The .editorConfig files are automatically found in the folder hierarchy of the
provided solution to be formatted.

adalon committed:
Removed rules that can be defined in the .editorConfig
